### PR TITLE
fix: change clock.timers from array to prototype-less object

### PR DESCRIPTION
**Pull request by Alexsey**
#### Purpose (TL;DR) - mandatory

Fix issue #159 - Use a faster (more time efficient) data structure for clock.timers

### CODE REVIEW

File: src/lolex-src.js [hunk 1]
```diff
@@ -231,7 +231,7 @@ function withGlobal(_global) {
         }
 
         if (!clock.timers) {
-            clock.timers = {};
+            clock.timers = Object.create(null);
```
Comment (benjamingr) [inline src/lolex-src.js]: Can you also change the other definition of `clock.timers`?

**Comment by benjamingr** [comment]
This means lolex can't support ES3 environments anymore (ES3 didn't have objects with null prototypes) - are those currently supported?

For the record - IE8 is the latest ES3 environment.

**Comment by benjamingr** [comment]
Also, it does not fix #159 - since you're still using an object rather than a PriorityQueue so everything is iterated every time a timer is looked for anyway.

**Comment by Alexsey** [comment]
About ES3 - Sinon says in its docs that [it's ES5.1](http://sinonjs.org/releases/v5.0.7/). There is nothing about supported ES version in lolex Readme so I decide that it is also targeting ES5.1

About priority queue - hashmap (object) is giving us [average O(1) on search, O(1) on insert and O(1) on delete](https://en.wikipedia.org/wiki/Hash_table) while all heaps would require [O(log n) at least on delete](https://en.wikipedia.org/wiki/Heap_(data_structure)). But if you think it could be faster - than ok, this is not closing #159 - it's just a bugfix

**Comment by benjamingr** [comment]
I don't think this change is a significant improvement to be honest since it doesn't actually make lolex faster. I don't feel strongly about this though.

**Comment by fatso83** [comment]
Yeah, I would guess you'd need to be in a pretty tight loop to ever notice the lack of guards in an actual scenario, but for what I know this might well be the case resulting in 5% improvement in some special case :man_shrugging: 

I don't think this is a significant improvement either, but if it's less code doing the same work in a clearer manner then it's usually a minor refactoring worth shoving in.

**Comment by benjamingr** [comment]
> Yeah, I would guess you'd need to be in a pretty tight loop to ever notice the lack of guards in an actual scenario,

`for...in` actually performs better with the `hasOwnProperty` check in V8 since V8 explicitly knows that construct. I'm not really up to date on what other browsers do though.

**Comment by Alexsey** [comment]
@fatso83 I can remove all `Object.create(null)` and leave only fix of initialization of `clock.timers` with array while it should be an object. In this case, would you accept this PR? Actually, that is a bug that doesn't affect logic but confuses anyone who reads the code

**Comment by fatso83** [comment]
Sure, I failed to see the fix in line 365.

**Comment by benjamingr** [comment]
@Alexsey we'd love to have more contributions from you to be clear - I hope the fact some of the changes here didn't make it wouldn't discourage you from working on more improvements :)

**Comment by Alexsey** [comment]
@benjamingr I understand that not every PR should be accepted - it's absolutely ok as long as there is some feedback. So here everything is ok :)

**Pull request closed without merging**